tests/test_cli.py

```diff
@@ -293,6 +299,48 @@ def test_correct_place_of_symlink_after_creation_of_it(self, mock_isfile):
              call(dst='/dst_folder/new_loc_add', src='/new_loc_add')],
             mock_symlink.call_args_list)
 
+    def test_delete_all_files_and_symlinks_in_destination_folder(self):
+        """ delete function should delete both file(s) and symlink(s) in destination folder. In this
+        test a temporary file and a symlink of it is created. Both of them should be deleted."""
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            with tempfile.NamedTemporaryFile(dir=tmpdirname) as tmpfile:
+                os.symlink(src=tmpfile.name, dst=tmpfile.name + '_symlink')  # symlink creation
+                sys.argv = [
+                    "",
+                    '-b', "2038-06-01",
+                    '-e', "2038-06-09",
+                    '-d', tmpdirname,
+                    "--time_to_live", "0"
+                ]
+                with mock.patch('shutil.copy'):
+                    with mock.patch('os.remove') as mock_os_remove:
+                        cli_copy.main()
+            os.remove(tmpfile.name + '_symlink')
+        self.assertCountEqual([call(tmpfile.name), call(tmpfile.name + '_symlink')],
+                              mock_os_remove.call_args_list)
+
+    @mock.patch('shutil.copy')
+    def test_deleting_symlinks_without_a_reference_file(self, mock_copy):
+        """ delete function should delete the symlink(s) regardless of the state of existence of the
+        reference file. In this test, the symlink is existing without a reference file. The test
+        shall remove that symlink."""
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            path_of_file = os.path.join(tmpdirname, 'test_file_name')
+            Path(path_of_file).touch()
+            symlink_path = path_of_file + '_symlink'
+            os.symlink(src=path_of_file, dst=symlink_path)  # symlink creation
+            os.remove(path_of_file)  # remove the reference file of the symlink
```

Comment:
Move that to the test above. Let this one test have three cases: one file, one valid symlink, one invalid symlink.

Reply:
I have done that, the mocks become confusing and the tests won't run correctly, it is better to test them separately:
![image](https://user-images.githubusercontent.com/62345618/104295379-ab572080-54c0-11eb-9b54-a0959da2e688.png)

![image](https://user-images.githubusercontent.com/62345618/104295417-ba3dd300-54c0-11eb-98f2-330ab3b7c50d.png)